src/plugins/explore/public/components/tabs/action_bar/patterns_settings/patterns_settings_popover_button.tsx

```diff
@@ -3,26 +3,39 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { EuiButtonIcon, EuiPopover } from '@elastic/eui';
+import { EuiButtonEmpty, EuiPopover } from '@elastic/eui';
 import React, { useState } from 'react';
+import { FormattedMessage } from '@osd/i18n/react';
+import { useSelector } from 'react-redux';
 import { PatternsSettingsPopoverContent } from './patterns_settings_popover_content';
+import { selectPatternsField } from '../../../../application/utils/state_management/selectors';
 
 export const PatternsSettingsPopoverButton = () => {
   const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const patternsField = useSelector(selectPatternsField) ?? '';
```

Comment:
what about if not found just the first field of type keyword you find on the dataset

Reply:
from the other discussion thread on the same line, we're going to have the selector read out a 'call to action' in the unlikely case that the field is not found.